app/models/concerns/courses/edit_options/qualification_concern.rb

```diff
@@ -16,9 +16,13 @@ def qualification_options
         def qualifications_with_qts
           qts_list = Course.qualifications.keys.grep(/qts/)
 
-          return qts_list if tda_active?
+          qts_list -= %w[undergraduate_degree_with_qts] if !tda_active? || non_tda_published?
 
-          qts_list - %w[undergraduate_degree_with_qts]
+          qts_list
```

Comment:
I really wish one dat to refactor the whole Edit options code because at least to me sounds wrong changing this module to remove an option in the view. And changing this module will affect different screens that uses the edit options.

I think the EditOption abstraction is bringing more problems than solutions. But maybe is just me. 

To solve this bug I have to add the condition above.